### PR TITLE
DOCS-1430: Add missing DataClient methods

### PR DESCRIPTION
**Pull request by andf-viam**
Add three missing DataClient API methods to DataClient API reference page:
- `add_bounding_box_to_image_by_id`
- `remove_bounding_box_from_image_by_id`
- `streaming_data_capture_upload`

### CODE REVIEW

File: docs/program/apis/data-client.md [hunk 1]
```diff
@@ -703,3 +703,97 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
 
 {{% /tab %}}
 {{< /tabs >}}
+
```
Comment (npentrel) [inline docs/program/apis/data-client.md:706]: Don't forget to also update these in `/static/include/services/apis/data-client.md`

Comment (andf-viam) [reply docs/program/apis/data-client.md:706]: Thank you!

**Review by stuqdog — APPROVED** [review]
LGTM!

**Comment by andf-viam** [comment]
@stuqdog Thanks so much!!

File: docs/build/program/apis/data-client.md [hunk 2]
```diff
@@ -634,6 +634,57 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+Upload the contents of streaming binary data and the relevant metadata to the [Viam app](https://app.viam.com).
+Uploaded streaming data can be found under the [**Data** tab](https://app.viam.com/data).
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `data` [(bytes)](https://docs.python.org/3/library/stdtypes.html#bytes-objects): the data to be uploaded, represented in bytes.
```
Comment (skyleilani) [inline docs/build/program/apis/data-client.md:647]:
```suggestion
- `data` [(bytes)](https://docs.python.org/3/library/stdtypes.html#bytes-objects): Data to be uploaded, represented in bytes.
```
Suggestion for uniformity with other parameter descriptions

File: docs/build/program/apis/data-client.md [hunk 3]
```diff
@@ -634,6 +634,57 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+- `component_name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Optional name of the component associated with the file.
+- `method_name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Optional name of the method associated with the file.
+- `method_parameters` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Optional dictionary of the method parameters. No longer in active use.
+- `data_request_times` [(Optional[Tuple[datetime.datetime, datetime.datetime]])](https://docs.python.org/3/library/stdtypes.html#tuples): Optional tuple containing [`datetime`](https://docs.python.org/3/library/datetime.html) objects denoting the times this data was requested and received by the appropriate sensor.
+- `tags` [(Optional[List[str]])](https://docs.python.org/3/library/stdtypes.html#typesseq-list): Optional list of [image tags](/data/dataset/#image-tags) to allow for tag-based data filtering when retrieving data.
+
+**Returns:**
+
+- [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): the `file_id` of the uploaded data.
```
Comment (skyleilani) [inline docs/build/program/apis/data-client.md:659]:
```suggestion
- [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The `file_id` of the uploaded data.
```
Same suggestion as above!

File: docs/build/program/apis/data-client.md [hunk 4]
```diff
@@ -705,3 +756,89 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
 
 {{% /tab %}}
 {{< /tabs >}}
+
+### AddBoundingBoxToImageById
+
+Add a bounding box to an image specified by its id.
```
Comment (skyleilani) [inline docs/build/program/apis/data-client.md:762]:
```suggestion
Add a bounding box to an image by its specified id.
```
Just a suggestion!

File: docs/build/program/apis/data-client.md [hunk 5]
```diff
@@ -705,3 +756,89 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/app/data_client/index.html#viam.app.data_client.DataClient.add_bounding_box_to_image_by_id).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### RemoveBoundingBoxFromImageById
+
+Removes a bounding box from an image specified by its id.
```
Comment (skyleilani) [inline docs/build/program/apis/data-client.md:812]:
```suggestion
Removes a bounding box from an image by its specified id.
```
Same suggestion as above

File: static/include/services/apis/data-client.md [hunk 6]
```diff
@@ -16,5 +16,8 @@ Method Name | Description
...
 [`BinaryDataCaptureUpload`](/build/program/apis/data-client/#binarydatacaptureupload) | Upload binary data collected on your machine through a specific component and the relevant metadata to the Viam app.
 [`TabularDataCaptureUpload`](/build/program/apis/data-client/#tabulardatacaptureupload) | Upload tabular data collected on your machine through a specific component and the relevant metadata to the Viam app.
+[`StreamingDataCaptureUpload`](/build/program/apis/data-client/#streamingdatacaptureupload) | Upload the contents of streaming binary data and the relevant metadata to the Viam app.
 [`FileUpload`](/build/program/apis/data-client/#fileupload) | Upload file data stored on your machine and the relevant metadata to the Viam app.
 [`FileUploadFromPath`](/build/program/apis/data-client/#fileuploadfrompath) | Upload file data stored on your machine from the specified filepath and the relevant metadata to the Viam app.
+[`AddBoundingBoxToImageById`](/build/program/apis/data-client/#addboundingboxtoimagebyid) | Add a bounding box to an image specified by its id.
```
Comment (skyleilani) [inline static/include/services/apis/data-client.md:22]:
```suggestion
[`AddBoundingBoxToImageById`](/build/program/apis/data-client/#addboundingboxtoimagebyid) | Add a bounding box to an image by its specified id.
```
Same suggestion as before

**Review by skyleilani — APPROVED** [review]
This looks really good to me!

File: docs/build/program/apis/data-client.md (hunk 4 above)
Comment (andf-viam) [reply docs/build/program/apis/data-client.md:762]: Ooh, this is tricky -- original way ties `its id` to `image` a bit more clearly, I worry that the rewrite ties it back to `box` erroneously. Definitely tricky either way.

But these should all be something better than `id` anyways, update everywhere to be `specified by its BinaryID`.

**Comment by andf-viam** [comment]
@sky Thanks for the helpful comments!!

File: docs/build/program/apis/data-client.md (hunk 4 above)
Comment (skyleilani) [reply docs/build/program/apis/data-client.md]: This makes a lot of sense, thank you for the explanation

**Pull request merged**